src/app/zap-templates/templates/app/shared-cluster-structs.zapt

```diff
@@ -2,15 +2,15 @@
 #pragma once
 
 #include <app/data-model/DecodableList.h>
-#include <app/data-model/Decode.h>
-#include <app/data-model/Encode.h>
 #include <app/data-model/List.h>
-#include <app/data-model/NullObject.h>
-#include <app/ConcreteAttributePath.h>
-#include <app/EventLoggingTypes.h>
-#include <lib/core/ClusterEnums.h>
-#include <lib/support/BitMask.h>
-#include <protocols/interaction_model/Constants.h>
+#include <lib/core/CHIPError.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/core/Optional.h>
```

Comment:
Should include BitMask.h.  And Nullable.h....  We might actually be missing Nullable.h above in some of the other bits too.

Maybe we need a new header that pulls in "all the types needed to represent things" as a followup and stop worrying about it in this PR, then update all these files to pull in that header?

Reply:
I am thinking we will pull things when things stop compiling. 
Creating a separate "include all" sounds like it creates a dumping ground... not that it is not something like that already as common includes are very much the same. 

I would be willing to see if we find something better in a followup. For now the bar seems to be "keep things compiling"

Reply:
> I am thinking we will pull things when things stop compiling.

The problem is this can break random people who just add a new cluster and suddenly it all blows up on them in a way they don't understand.

Reply:
I did give it a decent shot: I opened LSP on UnitTesting cluster because it is the largest ones (so most dependencies) and made sure it is without errors and even included some things from transitive includes.

Right now we are still mostly using `cluster-objects` which includes everything so we likely get transitive includes anyway and things compile. We may get more granular as we remove cluster dependencies ... I am proposing to fix it then. 

Alternatively I can try to compile every header stand-alone and if we pass compilation for the 100 clusters we have, we probably have decent coverage.

Reply:
Ah, I guess the PR go updated to add the missing headers in general.  OK, this is good to goo.